lib/blocs_and_cubits/user_cubit.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:game_lib_app/models/game/game.dart';
+import 'package:game_lib_app/models/user.dart';
+import 'package:game_lib_app/repositories/firestore_repository.dart';
+
+enum UserListNames {
+  favGames,
+  playedGames,
+  wishlistGames,
+}
+
+class UserCubit extends Cubit<User> {
+  final FirestoreRepository firestoreRepository;
+  UserCubit({
+    required this.firestoreRepository,
+  }) : super(User.initial());
+
+  Future<void> getUser(String? uid) async {
+    if (uid != null) {
+      emit(await firestoreRepository.getProfile(
+          uid: uid)); //metoda niedodajaca do bazy
+    }
+  }
```

Comment:
I suppose that this func should be in auth/user repository. Here it should be something like that: 
```suggestion
  emit(UserState.loading)....
  try => // func from repostiory getProfile
  emit(success)
  catch => emit(failed)
```